rest_framework/authentication.py

```diff
@@ -49,6 +49,32 @@ def authenticate_header(self, request):
         """
         pass
 
+    #: Name of openapi security scheme. Override if you want to customize it.
+    openapi_security_scheme_name = None
+
+    @classmethod
+    def openapi_security_scheme(cls):
+        """
+        Override this to return an Open API Specification `securityScheme object
+        <http://spec.openapis.org/oas/v3.0.3#security-scheme-object>`_
+        """
+        return {}
```

Comment:
TBH I would rather see this sort of hook added via a decorator (or such) rather than added in as part of the DRF API.

Reply:
@carltongibson Could you provide an example of how that might work? I don't quite understand.